tools/integration-tests/tester/tests/testutil.go

```diff
@@ -231,6 +233,59 @@ func SendFaucetRequest(t *testing.T, node *framework.Node, addr ledgerstate.Addr
 	return resp.ID, sent
 }
 
+// region CreateTransaction from outputs //////////////////////////////
+
+/**
+CreateTransactionFromOutputs takes the given utxos inputs and create a transaction tnat spread the total input balance
+across the targetAddresses. In order to correctly sign we have akeyPair map that maps a given address to its public key.
+Access and Consensus Mana is pledged to the node we specify.
+*/
+func CreateTransactionFromOutputs(t *testing.T, manaPledgeID identity.ID, targetAddresses []*ledgerstate.Address, keyPairs map[string]*ed25519.KeyPair, utxos ...ledgerstate.Output) *ledgerstate.Transaction {
+	// Create Inputs from utxos
+	inputs := ledgerstate.Inputs{}
+	balances := map[ledgerstate.Color]uint64{}
+	for _, output := range utxos {
+		output.Balances().ForEach(func(color ledgerstate.Color, balance uint64) bool {
+			balances[color] += balance
+			return true
+		})
+		inputs = append(inputs, output.Input())
+	}
+
+	// create outputs for each target address
+	numberOfOutputs := len(targetAddresses)
+	outputs := ledgerstate.Outputs{}
+	for i := 0; i < numberOfOutputs; i++ {
+		outBalances := map[ledgerstate.Color]uint64{}
+		for color, balance := range balances {
+			outBalances[color] = balance / uint64(numberOfOutputs)
+			if i == numberOfOutputs-1 {
+				outBalances[color] += balance % uint64(numberOfOutputs)
+			}
+		}
+		output := ledgerstate.NewSigLockedColoredOutput(ledgerstate.NewColoredBalances(outBalances), *targetAddresses[i])
+		outputs = append(outputs, output)
+	}
+
+	// create tx essence
+	txEssence := ledgerstate.NewTransactionEssence(0, time.Now(), manaPledgeID,
+		manaPledgeID, inputs, ledgerstate.NewOutputs(outputs...))
+
+	// create signatures
+	unlockBlocks := []ledgerstate.UnlockBlock{}
+	for i := 0; i < len(inputs); i++ {
```

Comment:
Same here: Since we know the length and iterate over the index it'd be better to initialize with make imo and then assign to the corresponding index instead of using append.